starlette/concurrency.py

```diff
@@ -50,8 +50,9 @@ def _next(iterator: typing.Iterator[T]) -> T:
 
 
 async def iterate_in_threadpool(
-    iterator: typing.Iterator[T],
-) -> typing.AsyncIterator[T]:
+    iterable: typing.Iterable[T],
+) -> typing.AsyncIterable[T]:
```

Comment:
Isn't the return an `AsyncIterator`?

Reply:
Good catch, thanks. Implemented this just now